File: Exercises/reverse.rb
def reverse(string)
  string_array = string.split("")
  size = string_array.length
  reversed = Array.new(size=size)

  current = size - 1
  while current >= 0 do
    reversed << string_array[current]
    current -= 1
  end

  reversed.join("")
end

puts reverse("cat rat bat")

File: Exercises/reverse.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define HELLO(i) i+5

void reverse(char s[]) {
  char *right_pointer = &s[strlen(s) - 1];
  char *left_pointer = &s[0];
  char placeholder;

  while(right_pointer > left_pointer)
  {
    placeholder = *left_pointer;
    *left_pointer = *right_pointer;
    *right_pointer = placeholder;

    right_pointer--;
    left_pointer++;
  }
}

int main(){
  char s[] = "cat rat bat !!!";

  reverse(s);
  printf("%s\n", s);

  return 0;
};
